src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -312,6 +323,60 @@ class EthService(
     }
   }
 
+  def getBalance(req: GetBalanceRequest): ServiceResponse[GetBalanceResponse] = {
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        GetBalanceResponse(account.balance)
+      }
+    }
+  }
+
+  def getStorageAt(req: GetStorageAtRequest): ServiceResponse[GetStorageAtResponse] = {
+    import io.iohk.ethereum.rlp.UInt256RLPImplicits._
+
+    val byteArrayUInt256Serializer = new ByteArrayEncoder[UInt256] {
+      override def toBytes(input: UInt256): Array[Byte] = input.bytes.toArray[Byte]
+    }
+
+    val rlpUInt256Serializer = new ByteArraySerializable[UInt256] {
+      override def fromBytes(bytes: Array[Byte]): UInt256 = ByteString(bytes).toUInt256
+      override def toBytes(input: UInt256): Array[Byte] = input.toBytes
+    }
+
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        val storageMpt =
+          MerklePatriciaTrie[UInt256, UInt256](account.storageRoot.toArray[Byte],
+            blockchainStorages.nodeStorage, crypto.kec256(_: Array[Byte]))(HashByteArraySerializable(byteArrayUInt256Serializer), rlpUInt256Serializer)
+        GetStorageAtResponse(storageMpt.get(UInt256(req.position)).getOrElse(UInt256(0)).bytes)
+      }
+    }
+  }
+
+  def getTransactionCount(req: GetTransactionCountRequest): ServiceResponse[GetTransactionCountResponse] = {
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        GetTransactionCountResponse(account.nonce)
+      }
+    }
+  }
+
+  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): Either[JsonRpcError, T] = {
+    withStateMpt(blockParam) { stateMpt =>
+      val account = stateMpt.get(kec256(address.toArray)).getOrElse(Account.Empty)
+      f(account)
+    }
+  }
+
+  private def withStateMpt[T](blockParam: BlockParam)(f: MerklePatriciaTrie[Array[Byte], Account] => T): Either[JsonRpcError, T] = {
+    import MerklePatriciaTrie.defaultByteArraySerializable
+    resolveBlock(blockParam).map { block =>
+      val stateMpt = MerklePatriciaTrie[Array[Byte], Account](block.header.stateRoot.toArray[Byte],
+        blockchainStorages.nodeStorage, (input: Array[Byte]) => crypto.kec256(input))
```

Comment:
How about we just define it in MPT companion obj? So that we can obtain it using just:
```
val mpt = MerklePatriciaTrie.stateMpt(block.stateRootHash, nodeStorage)
mpt.get(address)
...
```